File: app/controllers/MainController.java
package controllers;

import play.mvc.Controller;
import play.mvc.Result;
import javax.inject.Inject;
import javax.inject.Singleton;

@Singleton
public class MainController extends Controller {

    @Inject
    public MainController() {
    }

    public Result index() {
        return ok(views.html.index.render());
    }

}

File: server/app/controllers/MainController.scala
package controllers

import play.api.mvc._
import org.webjars.play.WebJarsUtil
import javax.inject._

@Singleton
class MainController @Inject()(
    val controllerComponents: ControllerComponents,
    implicit val webJarsUtil: WebJarsUtil
) extends BaseController {

  def index = Action {
    Ok(views.html.index())
  }

}
